geps/gep-2162.md

```diff
@@ -0,0 +1,112 @@
+2. Update the `GatewayClassStatus` struct to include `GatewayClassFeatures`
+
+```go
+// FeaturesSupport contains the features supported by the installed GatewayClass.
+type FeaturesSupport struct {
+    SupportedFeatures []string `json:"supportedFeatures,omitempty"`
+}
+
+// GatewayClassFeatures is the features supported by the GatewayClass broken down by support level.
+type GatewayClassFeatures struct {
+    // Core indicates the core support level features.
+    Core FeaturesSupport `json:"core,omitempty"`
+
+    // Extended indicates the extended support level features.
+    Extended FeaturesSupport `json:"extended,omitempty"`
+
+    // ImplementationSpecific indicates the implementation-specific support level features.
+    ImplementationSpecific FeaturesSupport `json:"extended,omitempty"`
+}
+// GatewayClassStatus is the current status for the GatewayClass.
+type GatewayClassStatus struct {
+    // Conditions is the current status from the controller for
+    // this GatewayClass.
+    //
+    // Controllers should prefer to publish conditions using values
+    // of GatewayClassConditionType for the type of each Condition.
+    //
+    // +optional
+    // +listType=map
+    // +listMapKey=type
+    // +kubebuilder:validation:MaxItems=8
+    // +kubebuilder:default={{type: "Accepted", status: "Unknown", message: "Waiting for controller", reason: "Pending", lastTransitionTime: "1970-01-01T00:00:00Z"}}
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+    // Features is the features supported by the GatewayClass broken down by support level.
+    Features GatewayClassFeatures `json:"features,omitempty"`
+}
+```
+
+## Conformance Details
+
+(TBD)
+
+## Alternatives
+
+### Re-using ConformanceProfiles structs
+
+We could use the same structs as we do in conformance profile object, more specifically, the [ProfileReport](https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/apis/v1alpha1/profilereport.go#LL24C6-L24C19) struct.
+
+Though it would be nice to have only one place to update, these structs seems to include much more data relevant to the conformance report but not for our use case. 
+
+That said, conformance profiles are still at experimental stage, we could explore the option to create a shared struct that will be used both for the conformance reports and for the GatewayClass status.
+
+### Instruct users to read from the future conformance profiles report
+
+The current plan for conformance profiles is to also include centralized reporting. (specific details don't matter now)
+We could wait for this to be implemented and instruct users to read from that source to determine what features their installed GatewayClass support.
+
+This does not cover a future piece of work we want to implement which is to warn/block users from applying a Gateway API object if the installed GWC doesn't support it. (originally suggested in [#1804](https://github.com/kubernetes-sigs/gateway-api/issues/1804)). 
```

Comment:
I think this is now well defined in the Conformance Profiles GEP now, I'd recommend linking to that. I think the reasons for publishing this on GWC status are:

1. Possibility for automation to block or warn when unsupported features are used, regardless of GatewayClass. For example, the incoming gwctl tool could help with that and/or potentially a validating webhook.
2. Ability for users to select the right GatewayClass for their usage without needing to refer to documentation or conformance reports. 
3. Ability for implementers to be able to specify implementation-specific features to differentiate their GatewayClasses.

I think 1 is the primary use case, while 2 and 3 are more nice to have than anything.